### PR TITLE
Make some Analyzer classes more flexible.

### PR DESCRIPTION
**Pull request by fhwang**

### CODE REVIEW

**Comment by fhwang** [comment]
Hold off on review for now, there's a little more work necessary on this branch.

**Comment by fhwang** [comment]
K, this is now ready for review. In the interest of cleaning up work in platform, I ended up changing two classes here:

# Analyzer::Container#run

Now returns a result object

# Analyzer::Engines

* Renamed to `Analyzer::EnginesBuilder`
* is-a Array behavior has been removed in favor of a more vanilla `run` method.
* `run` accepts an optional engine class. This way collaborators can take advantage of its config parsing functionality but still use a different engine class.

All of these changes are forward compatible.

@codeclimate/review

File: lib/cc/analyzer/engines_builder.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 module CC
   module Analyzer
-    class Engines < Array
+    class EnginesBuilder
       def initialize(registry:, config:, container_label:, source_dir:, requested_paths:)
         super()
```
Comment (pbrisbin) [inline lib/cc/analyzer/engines_builder.rb]: Probably don't need this now that you aren't subclassing `Array`

**Comment by pbrisbin** [comment]
LGTM

**Pull request merged**